devicesetup/src/main/java/io/particle/android/sdk/devicesetup/SetupCompleteIntentBuilder.java

```diff
@@ -0,0 +1,9 @@
+package io.particle.android.sdk.devicesetup;
+
+import android.content.Context;
+import android.content.Intent;
+import android.support.annotation.Nullable;
+
+public interface SetupCompleteIntentBuilder {
+    Intent buildIntent(Context ctx, @Nullable SetupResult result);
```

Comment:
I just realized that an abstract class still couldn't provide the `Parcelable.Creator`field; users would have to define that.  That extra degree of required boilerplate/complexity feels like a deal breaker.

Much as it kills me to say it, I think I just have to be OK with the fact that a whole bunch of SDK users are going to leak here.  I have some more ideas about improving this going forward (e.g.: an Android Lint rule to check for the leak, maybe based on this one: https://goo.gl/nWV6Gj ), but for now, this is what makes sense.